ksql-common/src/main/java/io/confluent/ksql/function/UdfFactory.java

```diff
@@ -154,53 +154,84 @@ private static boolean checkParamsMatch(final List<FunctionParameter> functionAr
   private List<FunctionParameter> mapToFunctionParameter(final List<Schema> params) {
     return params
         .stream()
-        .map(schema -> schema == null
-            ? new FunctionParameter(null, false)
-            : new FunctionParameter(schema.type(), schema.isOptional()))
+        .map(schema -> new FunctionParameter(schema))
         .collect(Collectors.toList());
   }
 
   private static final class FunctionParameter {
-    private final Schema.Type type;
-    private final boolean isOptional;
+    private final Schema schema;
 
-    private FunctionParameter(final Schema.Type type, final boolean isOptional) {
-      this.type = type;
-      this.isOptional = isOptional;
+    private FunctionParameter(final Schema schema) {
+      this.schema = schema;
     }
 
+    public boolean sameSchema(final Schema thatSchema) {
+      if (schema.type() != thatSchema.type()) {
+        return false;
+      }
+
+      if (schema.type().equals(Schema.Type.ARRAY)) {
```

Comment:
Can we add some tests to `UdfFactoryTest` to prove that we can indeed get functions with array/map and different generic types?

Reply:
I added to tests `shouldNotThrowExceptionForFunctionsWithDifferentTypeParametersForArray` and `shouldNotThrowExceptionForFunctionsWithDifferentTypeParametersForMap` for this reason.